Updates Backup launch price UX

ChaosExAnima opened the pull request:
#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* This updates the promo copy and styling for the new JP Backup plans.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* 1142395350490785-as-1152991457460141

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to My Plans page without the backup plan.
* Compare JP feature area to design.

Note: Functionality for switching between yearly and monthly states is not in scope of this PR.

Before (desktop): https://d.pr/i/R5KeBx
Before (mobile): https://d.pr/i/97e3qz

After (desktop): https://d.pr/i/qnMG5O
After (mobile): https://d.pr/i/amep9G

After, monthly: https://d.pr/i/psGySU

#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->
* N/A, part of wider JP Backup release notes.

ChaosExAnima committed:
First pass at updating UX.

ChaosExAnima committed:
Reverts changes in previous commit

ChaosExAnima committed:
Breaks out upgrade button

ChaosExAnima committed:
Fixes formatting of purchase button

ChaosExAnima committed:
Removes unneeded variable.

ChaosExAnima committed:
Loads star

ChaosExAnima committed:
Styles promo star

jeherve commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -29,13 +36,30 @@ export function SingleProductBackup( { plan, products, upgradeLinks, isFetchingD
+			<div className="single-product-backup__promo-star">{ __( 'Up to 70% off!' ) }</div>
+			<h4 className="single-product-backup__promo-header">
+				{ __( 'Hurry, these are' ) }
+				<strong>{ __( 'Limited time introductory prices!' ) }</strong>
```
This is going to be problematic for translators. I think it would be best to have this be one string only.

ChaosExAnima replied:
Combined in 4fc316f6607456a9ab22642a9d001203dd4d7b01

jeherve commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -155,6 +173,47 @@ function PlanRadioButton( {
+				{ __( 'Upgrade to %(name) for %(price)/per %(billingTime)', {
+					args: {
+						name,
+						billingTime: billingTimeFrameString,
```
This may be complicated for translators as well. Even if we were to make `billingTimeFrameString` translatable when we define it above, translators won't always be able to build a nice phrase for it. Indeed, in some languages "per year" is one word only, and is different from "year".

ChaosExAnima replied:
Tweaked in a548f653366f1ebe2aab681c6fdf702f9cc7ca2d

jeherve commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -155,6 +173,47 @@ function PlanRadioButton( {
+		discountedPrice || fullPrice,
+		currencyCode
+	);
+	const price = `${ symbol }${ integer }${ fraction }`;
```
Any chance we could make sure this is localized? Not all languages use the currency code before the price, for example.

ChaosExAnima replied:
I'd be super game, but I didn't see that anywhere and the `PlanPrice` component hardcoded it the same way.

ChaosExAnima replied:
@jeherve - So I did some digging, and found that the [currency formatting library](https://github.com/Automattic/wp-calypso/blob/master/packages/format-currency/src/index.js) does not support currencies with the code after the price. I've opened [an issue](https://github.com/Automattic/wp-calypso/issues/38325) and can resolve when I get a moment- it also seems the NPM package is out of date there too.

ChaosExAnima replied:
As an update: I added [a placeholder function](https://github.com/Automattic/jetpack/commit/61f85718cf3a715b1b85a13df7ab4883095d6c67#diff-7457a6cdabf24fa36923c20d4c9b101cR254) for this, and will work on fixing the currency formatting issue and retroactively patch this to remove my stub function. Just don't want it blocking things!

jeherve commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -238,19 +274,22 @@ class SingleProductBackupBody extends React.Component {
-					) }
+				<p>
+					<em>
+						{ __( 'You are saving $40 by paying yearly. {{a}}Switch to monthly{{/a}}', {
```
Is it okay to have the `$40` value be static here? Wouldn't we want this to be dynamic so it is always accurate, regardless of ongoing sales or currency?

tyxla replied:
FWIW, if we take Jetpack Backup Real-time, here's how we calculate the savings in various locations (I'm using Euros for the example, but it's the same for any currency, and you get the picture):

* Monthly price - €20 per month
* Yearly price: €200 per year
* X = Price for a year, when paying monthly - 12 * €20 = €240
* Y = Price for a year, when paying yearly - €200
* Savings when paying yearly: X - Y = €240 - €200 = €40

So we can totally keep it dynamic.

This calculation already exists in other locations, so ideally we shouldn't be duplicating it.

ChaosExAnima replied:
I adjusted things to calculate the difference and set text appropriately in 61f85718cf3a715b1b85a13df7ab4883095d6c67. Sorry for the large and vague commit :)

jeherve commented on _inc/client/plans/single-product-backup.scss:
```diff
@@ -125,6 +138,34 @@ input[type='radio'].plan-radio-button__input {
+}
+
+.single-product-backup__promo {
+	background-color: #f8f8f8;
```
Should we have the color here and below be pulled from a variable?

ChaosExAnima replied:
Sure! Should I add the variable to the global `colors.scss`, or limit it to this file?

jeherve replied:
I think `colors.scss` would make sense 👍

ChaosExAnima replied:
Cool! I fixed in 431388873810fcdaf39b9ab6fc3e2417a190de59, along with tweaking a color to match other headers on page from WP Core admin colors.

ChaosExAnima committed:
Moves component into translated string.

ChaosExAnima committed:
Adds additional context for translaters

ChaosExAnima committed:
Moves colors to central file.

ChaosExAnima committed:
Renames variables to be clearer.

ChaosExAnima committed:
Cleans up data.

ChaosExAnima committed:
Moves prop into parent and adds clarifying comment.

ChaosExAnima committed:
Reverts change to Yarn.

ChaosExAnima committed:
Changes time frame back to yearly.

ChaosExAnima commented:
Fixed all issues flagged for me, let me know if there's anything missing or more optimal ways to structure things. Thanks!

delawski commented on _inc/client/plans/single-product-backup.scss:
```diff
@@ -89,11 +93,23 @@ input[type='radio'].plan-radio-button__input {
+
+		&::before {
+			display: block;
+			border-left: 1px solid $gray-text-min;
```
It feels that the border is too dark compared to mockups. Could we maybe use here some a lighter grey swatch?

ChaosExAnima replied:
Switched it up in 9a8cc1dc783be55e20acb1ac58c2ddd824b89ef7, think it looks much better now!

delawski commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -29,56 +30,86 @@ export function SingleProductBackup( { plan, products, upgradeLinks, isFetchingD
+	return (
+		<div className="single-product-backup__promo">
+			<div className="single-product-backup__promo-star">
+				<img src={ `${ imagePath }/green-star.svg` } alt="" />
```
Since this is a purely decorative image used as a text background, could we define it in the CSS using `background-image` property?

ChaosExAnima replied:
We could! I've seen it both ways in the code. What's the best practice here?

delawski replied:
I guess that the general best practice is not to use `img` for presentational elements if possible.

If we want to keep the `img` tag here, though, let's maybe address @tyxla's suggestion and add the `role="presentation"` here.

ChaosExAnima replied:
That makes sense. I've switched to a background image in 8ea1461f4cf65e613c0536b64f0f2a1b3cc5d3a8, which also allowed the markup to be simpler.

delawski commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -155,15 +186,93 @@ function PlanRadioButton( {
+		return null;
+	}
+	return (
+		<p>
```
I guess that - since those HTML elements are common to both saving text variations - we could move them to the parent component. This way the `PlanSavingsText` function could be significantly simplified by returning one of the translations early.

ChaosExAnima replied:
Done in a89598e6e!

delawski commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -155,15 +186,93 @@ function PlanRadioButton( {
+	return (
+		<div className="single-product-backup__upgrade-button-container">
+			<Button href={ link } onClick={ onClickHandler( type ) } primary>
+				{ __( 'Upgrade to %(name)s for %(price)s/%(billingTime)s', {
```
Even though the designs show the slash (`/`) here, I believe it's redundant. The time frame copy already states it is "per [time period]".

ChaosExAnima replied:
Done in ae5bde1c126081336373f935b81bc3269b94bee2!

delawski commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -155,15 +186,93 @@ function PlanRadioButton( {
+					args: {
+						name,
+						billingTime: billingTimeFrameString,
+						price: formatCurrency( currencyObject ),
```
Could we, by any chance, reuse here the [`PlanPrice` component](https://github.com/Automattic/jetpack/blob/2461484dad9f4bfa8f963552f0a6235920939123/_inc/client/components/plans/plan-price/index.jsx)? We would have all the currency handling out-of-the box.

For sure one obstacle is the fact that the `PlanPrice` returns `h4` element and we want to use it here inside a `button`. It'd have to be modified to return just a `span`, for instance. Alternatively, it could accept an argument to tell if it should return a `h4` tag or just a `Fragment` instead, giving us a full flexibility.

What do you think?

ChaosExAnima replied:
I think modifying the `PlanPrice` component would probably be the way to go, honestly. I didn't use it because of the `h4`, but you're right that adding a prop there would allow us to reuse it.

delawski replied:
I went ahead and issued a PR that adds an `inline` option to the `PlanPrice` component (and also refactors it): #14231 

Please review and feel free to re-use it here now or after it's merged (if the #14231 gets into master after this one).

delawski reviewed (commented):
Great work, @ChaosExAnima!

I reviewed the code and left some suggestions. I don't there are any blockers, but I think they are worth considering in follow-up pushes.

tyxla commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -179,52 +288,20 @@ class SingleProductBackupBody extends React.Component {
 		return (
 			<React.Fragment>
-				<p>
-					{ __(
```
I see we removed this link, but I expected that we move it to the subheader, as shown in Figma.

ChaosExAnima replied:
So, the `onClick` handler here uses the currently selected plan, which is data only available in this component. If we'd like to add this link to the header, we'd need to not have the analytics click event called in `handleLandingPageLinkClick`. Is that acceptable?

delawski replied:
I think that to overcome this limitation we could keep the `selectedBackupType` value in the parent component's (`ProductSelector`) state where the header is defined.

delawski replied:
I've actually went ahead and issued a PR that does just that - moves the link to the parent component: #14229 

Please, review and - depending which PR gets merged first - rebase the PR. I'm happy to take care of that on Monday as well.

tyxla commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -179,52 +288,20 @@ class SingleProductBackupBody extends React.Component {
-					) }
-				</p>
-				<h4 className="single-product-backup__options-header">{ __( 'Backup options:' ) }</h4>
+				<p>{ __( 'Always-on backups ensure you never lose your site.' ) }</p>
```
I think we could use some padding below this one.

ChaosExAnima replied:
Addressed in f879054c0a1452457d34fb71bc02d744114fe4e6, but it doesn't look great to me?

tyxla replied:
I think you should add some more 😉 

Right now we have:

![](https://cldup.com/0qW0IhQ7bq.png)

And I think that in the design, the space above and below this paragraph is equal.

ChaosExAnima replied:
Okay, so I tweaked things in 131362ec28a949bbba7afbef9e15c6102fcd87cd:

![Screenshot on 2019-12-16 at 14_53_54](https://user-images.githubusercontent.com/1760168/70938482-3d69cc80-2014-11ea-86cf-b6304f83eece.png)

@tyxla - look good to you?

delawski replied:
@ChaosExAnima Thank you for refining the appearance of the card here.

I've got just one minor request: can you align the promo nudge copy to the left (for both mobile and desktop breakpoints) so that it matches the designs?

tyxla replied:
Left-aligned the copy in https://github.com/Automattic/jetpack/pull/14203/commits/af110037d68ed8cee1da576fc3c8a07876e22dc2.

Regarding the vertical space - for wp-admin it seems that design doesn't suggest additional vertical space - should we keep it as-is?

delawski replied:
To me it seems that there is much more space before and after the promo nudge in the designs, but I'm happy to leave it as is. I'm sure this will go through a testing/design review at some point so we can address minor issues like this at the time.

tyxla replied:
Thanks, I've added some more space below the promo nudge!

tyxla reviewed (commented):
This is working well, nice work @ChaosExAnima!

It's a big PR though, I would have loved it if we could some of the following in separate PRs:

* Billing timeframe handling and logic
* Componentization - moving/building functionality to subcomponents

I've also left some comments and questions, let me know what you think.

ChaosExAnima committed:
Removes logic for plan switching in favor of existing changes.

ChaosExAnima commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -29,56 +30,86 @@ export function SingleProductBackup( { plan, products, upgradeLinks, isFetchingD
 					billingTimeFrame={ billingTimeFrame }
 					currencyCode={ currencyCode }
-					backupOptions={ backupOptions }
+					backupOptions={ { realTime: realTimeOptions, daily: dailyOptions } }
```
Removed in d164367f319ecb0c5497d1654b88a35eb2dc3961

ChaosExAnima commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -155,15 +186,93 @@ function PlanRadioButton( {
+	const switchComponent = <a />;
+
+	let promptText;
+	if ( 'yearly' === billingTimeFrame ) {
```
I think that there's not really a way to have both separate PRs for logic and presentation here, as there's not a way for the component to alter the state as it's written.

What I'd suggest is perhaps to remove the `Switch to X` string entirely and implement that in our logic PR as an addition. Thoughts?

ChaosExAnima replied:
Given async nature of this conversation, I've simplified in c1541b5879fa5e424363c83f96d1b6154ba2de30. Let me know if you have further feedback!

delawski replied:
> What I'd suggest is perhaps to remove the Switch to X string entirely and implement that in our logic PR as an addition. Thoughts?

I like this approach, let's do that 👍

tyxla replied:
I was hoping we wouldn't be implementing this in this PR at all. It's a remnant of old designs and it should be removed for this iteration. It makes sense in a separate PR IMHO.

ChaosExAnima committed:
Simplifies savings components.

Does the following changes:
* Renames component to make more sense.
* Removes link for component as logic is not implemented.
* Removes billing timing switch.

ChaosExAnima commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -155,15 +186,93 @@ function PlanRadioButton( {
 }
 
-// eslint-disable-next-line no-unused-vars
+function PlanSavingsText( { selectedBackup, billingTimeFrame, currencyCode } ) {
```
Sounds great to me! Updated in c1541b5879fa5e424363c83f96d1b6154ba2de30

ChaosExAnima committed:
Changes promotional text to use percentage variable instead.

ChaosExAnima commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -29,56 +30,86 @@ export function SingleProductBackup( { plan, products, upgradeLinks, isFetchingD
+		<div className="single-product-backup__promo">
+			<div className="single-product-backup__promo-star">
+				<img src={ `${ imagePath }/green-star.svg` } alt="" />
+				<span className="single-product-backup__promo-star-text">{ __( 'Up to 70% off!' ) }</span>
```
Make sense! I am unused to actually having translations of things :sweat_smile: 

Updated in deda9468b

ChaosExAnima committed:
Alters ProductSavings component to only return text.

ChaosExAnima commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -155,15 +186,93 @@ function PlanRadioButton( {
+	const { link, name, fullPrice, discountedPrice, type } = selectedUpgrade;
+	let billingTimeFrameString = '';
+	if ( 'yearly' === billingTimeFrame ) {
+		billingTimeFrameString = __( 'per year', { context: 'Amount of money per time unit.' } );
```
Interesting question! I did a search on the codebase for `'per year'` but got only this file. Would there be other places to look?

delawski replied:
> Would there be other places to look?

We have something very similar in [`SingleProductBackupPriceGroup`](https://github.com/Automattic/jetpack/blob/8b9ab853823ea5f811c252a1e8f53b0bf8b79134/_inc/client/plans/single-product-backup.jsx#L140-L153). I wonder if we could reuse it here? But I also don't know if adding a `comment` or `context` introduces a new translation string.

ChaosExAnima replied:
Got clarification from i18n team, and we can find [additional info here](https://translate.wordpress.com/2016/11/24/context-vs-comment/). In short, use a comment, not a context, so this is absolutely reusable. Will update.

ChaosExAnima replied:
So I implemented it in 0cac84d0340e7e0ff50dc8c7ec72337b76a1c9f3, but as a note we'll need to wait for #14231 to allow this to be inline.

My inclination right now is to hold this until those two PRs are approved and merged, and then complete this one.

tyxla replied:
Sounds good! 👍

ChaosExAnima committed:
Removes superfluous slash in text.

ChaosExAnima committed:
Renames string replacement name.

ChaosExAnima commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -155,15 +186,93 @@ function PlanRadioButton( {
+	return (
+		<div className="single-product-backup__upgrade-button-container">
+			<Button href={ link } onClick={ onClickHandler( type ) } primary>
+				{ __( 'Upgrade to %(name)s for %(price)s/%(billingTime)s', {
```
Done in 91ac5d7b1e251b13fa0baeda9057ec3fc4f9775d

ChaosExAnima committed:
Reverts changes to real time variable name.

ChaosExAnima committed:
Simplified logic to avoid terniaries in string literals.

ChaosExAnima commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -155,15 +186,93 @@ function PlanRadioButton( {
+// until https://github.com/Automattic/wp-calypso/pull/36039
+// is released.
+function formatCurrency( { symbol, integer, raw, fraction } ) {
+	return `${ symbol }${ integer }${ raw - integer > 0 ? fraction : '' }`;
```
That makes a lot of sense! I've changed things in 649abd900aec29842b753c1266fae437ff128c95

ChaosExAnima committed:
Adds detail for translators.

ChaosExAnima committed:
Fixes formatting to display var correctly.

ChaosExAnima committed:
Removes invalid code.

The raw variable is not available, making there no way to get the fractional price.

ChaosExAnima commented:
Answered a few things. Will address last items tomorrow.

Thank you for the thorough review @delawski and @tyxla!

delawski commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -36,6 +37,28 @@ export function SingleProductBackup( { plan, products, upgradeLinks, isFetchingD
+			<div className="single-product-backup__promo-star">
+				<img src={ `${ imagePath }/green-star.svg` } alt="" />
+				<span className="single-product-backup__promo-star-text">
+					{ __( 'Up to %(percent)d%% off!', { args: { percent } } ) }
```
How about having this translation more flexible so that not only percentage values are accepted here?

We could do this by replacing the `percent` with a `discount` like:

```diff
- const percent = 70;
+ const discount = '70%';
```

and then the changing the translation to:

```diff
- { __( 'Up to %(percent)d%% off!', { args: { percent } } ) }
+ { __( 'Up to %(discount)s off!', { args: { discount } } ) }
```

ChaosExAnima replied:
@delawski - Do we know if i18n will affect the positioning of the percentage signs, or is it universally on the right side? I wrote it like that so the translators can move/change the percentage.

delawski replied:
You are right, @ChaosExAnima! I just read about [some rules around percentage sign](https://en.wikipedia.org/wiki/Percent_sign). There are cases when it can be on the other side of the number (e.g. Turkish, Persian, Arabic).

Please disregard my comment and leave the code as is.

ChaosExAnima commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -29,56 +30,86 @@ export function SingleProductBackup( { plan, products, upgradeLinks, isFetchingD
+	};
+
+	if ( 'yearly' === billingTimeFrame ) {
+		dailyOptions = {
```
Reverted these changes in d164367f319ecb0c5497d1654b88a35eb2dc3961

ChaosExAnima committed:
Uses color from Calypso SASS.

ChaosExAnima committed:
Uses background image instead of image tag.

ChaosExAnima commented on _inc/client/scss/variables/_colors.scss:
```diff
@@ -96,10 +96,13 @@ $green-primary:   #00BE28;
 $green-secondary: #00a523;
 $green-dark:      #008b1d;
 
+$color-off-white: #f8f8f8;
```
Good catch! Resolved in e4d30fbf731f94bf0abd33922431d2319dd9885f

ChaosExAnima commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -29,56 +30,86 @@ export function SingleProductBackup( { plan, products, upgradeLinks, isFetchingD
+	return (
+		<div className="single-product-backup__promo">
+			<div className="single-product-backup__promo-star">
+				<img src={ `${ imagePath }/green-star.svg` } alt="" />
```
Outdated due to 8ea1461f4cf65e613c0536b64f0f2a1b3cc5d3a8

ChaosExAnima committed:
Lightens border color.

ChaosExAnima committed:
Duplicates i18n text to prevent different strings

ChaosExAnima committed:
Increases button top margin.

tyxla commented on yarn.lock:
```diff
@@ -7488,11 +7488,6 @@ is-path-inside@^3.0.1:
   resolved "https://registry.yarnpkg.com/is-path-inside/-/is-path-inside-3.0.2.tgz#f5220fc82a3e233757291dddc9c5877f2a1f3017"
   integrity sha512-/2UGPSgmtqwo1ktx8NDHjuPwZWmHhO+gj0f93EkhLB5RgW9RZevWYYlIkS6zePc6U2WpOdQYIwHe9YC4DWEBVg==
 
-is-plain-obj@^1.0.0:
```
Ideally, let's keep those changes out of this PR and file them in a separate one.

ChaosExAnima replied:
Yep, accidentally pushed those. Fixed.

tyxla commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -155,15 +172,70 @@ function PlanRadioButton( {
+	if ( 'yearly' === billingTimeFrame ) {
+		billingTimeFrameString = __( '{{price/}} {{timeframe}}per year{{/timeframe}}', {
+			components: { price, timeframe },
+			comment:
```
Should we also explain that the `timeframe`'s purpose is purely cosmetic?

tyxla commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -155,15 +172,70 @@ function PlanRadioButton( {
+		<PlanPrice currencyCode={ currencyCode } rawPrice={ discountedPrice || fullPrice } inline />
+	);
+	if ( 'yearly' === billingTimeFrame ) {
+		billingTimeFrameString = __( '{{price/}} {{timeframe}}per year{{/timeframe}}', {
```
I'm not sure we should have that in a single string. In other places (Calypso), we always keep the price separate, and then we only have to translate `per year` and `per month`, and I think they already have translations.

This would also resolve another problem, where we have duplicate strings where only the timeframe is different.

tyxla commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -155,15 +172,70 @@ function PlanRadioButton( {
+					components: { billingTimeFrame: billingTimeFrameString },
+					args: { name },
+					comment:
+						'Button to purchase product upgrade. %(price) can be a range of prices, and %(billingTimeFrame) is the billing period for the product upgrade.',
```
This comment mentions a price placeholder, but I don't see one in this string.

tyxla commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -238,19 +278,21 @@ class SingleProductBackupBody extends React.Component {
+						/>
+					</em>
+				</p>
+				<UpgradeButton
```
Something seems off since we started using this component for the upgrade button:

![](https://cldup.com/rYNgAVSj_2.png)

delawski replied:
I guess this is because #14231 is not yet merged and so the `PlanPrice` disregards `inline` prop.

ChaosExAnima replied:
Yes, this should be resolved now that I've rebased!

tyxla reviewed (commented):
This is starting to get together nicely!

I think it will need a rebase once #14229 and #14231 land.

tyxla commented:
Changed the PR to "in progress" as it relies on some other PRs to be merged right now. Let's make it "needs review" once it's rebased and ready for review.

ChaosExAnima committed:
Resets yarn

ChaosExAnima committed:
Reverts merge conflicts.

ChaosExAnima committed:
Actually reverts merge conflicts.

ChaosExAnima committed:
Wires up state to product selector

ChaosExAnima committed:
Restructures i18n for product price button.

ChaosExAnima committed:
Fixes bug with string formatting.

ChaosExAnima committed:
Replicates formatting string for i18n

ChaosExAnima commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -108,20 +125,24 @@ function SingleProductBackupPriceGroup( {
+			{ price }
+			<div className="single-product-backup__price-group-billing-timeframe">
+				{ billingTimeFrameString }
+			</div>
```
@tyxla @delawski - How's this setup? I copied the  format exactly from [here](https://github.com/Automattic/jetpack/pull/14203/files#diff-7457a6cdabf24fa36923c20d4c9b101cR203-R210) to reduce translated strings, but I wasn't sure if having a string that's entirely placeholders is good for translation.

ChaosExAnima committed:
Changes margins per design feedback.

ChaosExAnima committed:
Pushes price processing to already made component

ChaosExAnima commented:
Okay, let's do another round! Changes this time:

- Rebased to latest code from master.
- Updated billing timeframe text to create simpler strings across both `UpgradeButton` and `SingleProductBackupPriceGroup`.
- Got rid of currency formatting in favor of universally relying on `PlanPrice` component.
- Wired up state to respect parent's backup state.
- Fixed margins.

Let me know if you have any other issues, @delawski and @tyxla!

tyxla commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -216,19 +278,21 @@ class SingleProductBackupBody extends React.Component {
-					) }
+				<p>
+					<em>
+						<ProductSavings
```
If we want to keep this here, we should make sure that the user can switch to monthly. Otherwise, it's weird that they're saving if they don't have the other option (monthly) available to them 😉

tyxla replied:
I guess, let's go with it for now, but consider implementing yearly/monthly billing switch in wp-admin for backups.

tyxla reviewed (changes requested):
This works well, @ChaosExAnima. Great work! 🙌 

I've left some last minute requests we should think about, but I think we should be good once we address them.

tyxla committed:
Use math only on numbers that exist

delawski commented:
Thank you for the updates, @ChaosExAnima! I left just a minor comment. Other than that (and Marin's feedback), I think it's good to go 👍

tyxla committed:
Extract billing timeframe logic to a function

tyxla committed:
Add a clarification for the missing arg in the localized string

tyxla committed:
Use CSS instead of em tag to make savings italic

tyxla committed:
Left-align the promo header

tyxla commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -55,27 +74,25 @@ function SingleProductBackupCard( {
+			link: upgradeLinks.daily,
 			discountedPrice: priceDaily,
 			fullPrice: priceDailyMonthlyPerYear,
+			potentialSavings: priceDailyMonthlyPerYear - priceDaily,
```
Done in https://github.com/Automattic/jetpack/pull/14203/commits/cc98b3e98ad4ff974f591f724534b041bf06f5e4.

tyxla commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -55,27 +74,25 @@ function SingleProductBackupCard( {
+			link: upgradeLinks[ 'real-time' ],
 			discountedPrice: priceRealtime,
 			fullPrice: priceRealtimeMonthlyPerYear,
+			potentialSavings: priceRealtimeMonthlyPerYear - priceRealtime,
```
Done in https://github.com/Automattic/jetpack/pull/14203/commits/cc98b3e98ad4ff974f591f724534b041bf06f5e4.

tyxla commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -158,15 +178,64 @@ function PlanRadioButton( {
+	const price = (
+		<PlanPrice currencyCode={ currencyCode } rawPrice={ discountedPrice || fullPrice } inline />
+	);
+	if ( 'yearly' === billingTimeFrame ) {
```
Done in https://github.com/Automattic/jetpack/pull/14203/commits/cc98b3e98ad4ff974f591f724534b041bf06f5e4.

tyxla commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -158,15 +178,64 @@ function PlanRadioButton( {
+					components: { price },
+					args: { name, billingTimeFrame: billingTimeFrameString },
+					comment:
+						'Button to purchase product upgrade. {{price /}} can be a range of prices, and %(billingTimeFrame) is the billing period for the product upgrade.',
```
Updated in https://github.com/Automattic/jetpack/pull/14203/commits/aff60c15dfe7110b7800ec21745ace0b5a184a9f.

tyxla commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -216,19 +278,21 @@ class SingleProductBackupBody extends React.Component {
-						</div>
-					) }
+				<p>
+					<em>
```
Addressed in https://github.com/Automattic/jetpack/pull/14203/commits/84426b63600eed9163cea98978ddd99531893a06.

tyxla commented:
Addressed all the feedback. This is ready for another round of reviews - @delawski mind taking another 👀 ?

tyxla committed:
Make ProductSavings self-sufficient

delawski reviewed (dismissed):
Everything looks good to me now! Let's 🚢it!

tyxla committed:
Add some space below promo section

tyxla reviewed (approved):
This looks good now. @delawski can you take a look and ✅ it too before 🚢 ?

delawski commented:
It's better now, @tyxla 👍Let's have it like this and ship!
